src/http/cookie.cr

```diff
@@ -217,83 +300,161 @@ module HTTP
+    # cookies["debug"]?   # => false
     # ```
-    def []?(key)
+    def []?(key : String) : Cookie?
       @cookies[key]?
     end
 
     # Returns `true` if a cookie with the given *key* exists.
     #
     # ```
-    # request.cookies.has_key?("foo") # => true
+    # require "http/cookie"
+    #
+    # cookies = HTTP::Cookies.new{HTTP::Cookie.new("session", "god")}
+    # cookies.has_key?("session") # => true
+    # cookies.has_key?("user")    # => false
     # ```
-    def has_key?(key)
+    def has_key?(key : String) : Bool
       @cookies.has_key?(key)
     end
 
-    # Adds the given *cookie* to this collection, overrides an existing cookie
+    # Adds the given *cookie* to this collection indexed as the name of the cookie.  If another cookie is added with the same name it will overwrite the older cookie.
     # with the same name if present.
     #
     # ```
-    # response.cookies << HTTP::Cookie.new("foo", "bar", http_only: true)
+    # require "http/cookie"
+    #
+    # cookies = HTTP::Cookies.new{HTTP::Cookie.new("session", "god")}
+    # cookies # => <HTTP::Cookies @cookies={"session" => <HTTP::Cookie @name="session", @value="god" ...>}>
+    # cookies << HTTP::Cookies.new("session", "master")
+    # cookies # => <HTTP::Cookies @cookies={"session" => <HTTP::Cookie @name="session", @value="master" ...>}>
     # ```
-    def <<(cookie : Cookie)
+    def <<(cookie : Cookie) : Cookie
       self[cookie.name] = cookie
     end
 
     # Clears the collection, removing all cookies.
-    def clear
+    #
+    # ```
+    # require "http/cookie"
+    #
+    # cookies = HTTP::Cookies.new{HTTP::Cookie.new("session", "god")}
+    # cookies.size # => 1
+    # cookies.clean
+    # cookies.size # => 0
+    # ```
+    def clear : Hash(String, Cookie)
       @cookies.clear
     end
 
     # Deletes and returns the `HTTP::Cookie` for the specified *key*, or
     # returns `nil` if *key* cannot be found in the collection. Note that
     # *key* should match the name attribute of the desired `HTTP::Cookie`.
-    def delete(key)
+    #
+    # ```crystal
+    # require "http/cookie"
+    #
+    # cookies = HTTP::Cookies.new{HTTP::Cookie.new("session", "god")}
+    # cookies.delete("session") # => <HTTP::Cookie @name="session" @value="god" ...>
+    # cookies.empty?            # => true
+    # ```
+    def delete(key : String) : Cookie?
       @cookies.delete(key)
     end
 
-    # Yields each `HTTP::Cookie` in the collection.
-    def each(&block : Cookie ->)
+    # Iterates over the headers yeilding each header as a `Tuple` of `String` and an `Array(String)`.
+    #
+    # ```
+    # require "http/cookie"
+    #
+    # cookies = HTTP::Cookies{HTTP::Cookie.new("session", "god")}
+    # cookies_hash = {} of String => String
+    #
+    # cookies.each do |cookie|
+    #   cookie    # => <HTTP::Cookie @name="session, @value="god" ...>
+    #   cookie_hash[cookie.name] = cookie.value
+    # end
+    #
+    # header_hash # => {"session" => "name"}
```

Comment:
You're using three different names for the collector variable 🤦

Reply:
copy pasting does not always work